### PR TITLE
[WIP] Update vim syntax highlighting to reflect current options

### PR DESCRIPTION
**Pull request by Tranquility**
Added:
- tnoremap
- tmap
- tunmap
- TermOpen

Removed
- termcap code options
- hardtabs
- compatible
- cp
- nocompatible
- nocp

### CODE REVIEW

**Comment by noahfrederick** [comment]
Looks like `tunmap` also needs highlighting.

**Comment by tarruda** [comment]
Thanks, can you also add for the `TermOpen` autocmd?

**Comment by Tranquility** [comment]
I updated the commit to contain `tunmap` and `TermOpen`.

**Comment by ZyX-I** [comment]
AFAIR things like options, commands and functions, not sure about autocommands were purely generated. How about removing these parts from the `syntax/vim.vim` file completely and add `syntax/vim/generated.vim` which will be generated as a part of the build process?

**Comment by Tranquility** [comment]
> How about removing these parts from the syntax/vim.vim file completely and add syntax/vim/generated.vim which will be generated as a part of the build process?

Sounds much better, editing these by hand is error prone and easy to forget. I just fear I don't have have enough knowledge to implement this.

**Comment by ZyX-I** [comment]
@Tranquility I always thought there *already* is a script somewhere that either does full update or, like with unicode script (see removed file in #2159) generates update which needs to be copied later by hand. (There were some references in vim-dev that such a thing exists, but I never saw a link and was not very interested in finding those.)

**Comment by Pyrohh** [comment]
@ZyX-I Are you referring to `mkvimvim` (#1680)? I'm guessing that it either hasn't been used in a long time, or was never in https://github.com/vim/vim, as `git log -Smkvimvim` only shows the same comment in `syntax/vim.vim` repeatedly.

**Comment by Pyrohh** [comment]
Unless someone plans on making such a script soon, then this seems like an adequate solution.

File: runtime/syntax/vim.vim [hunk 1]
```diff
@@ -38,31 +38,16 @@ syn keyword vimOption contained	altkeymap arabic autowrite backupcopy bdir bin b
...
 syn keyword vimOption contained	noai noaltkeymap noar noarabicshape noautochdir noautowrite noawa nobeval nobl nocf nocindent nocp nocscopetag nocst nocul nocursorline nodg noea noeol noesckeys noexpandtab nofic nofkmap nogdefault nohidden nohkmapp nohls noic noignorecase noimc noimd noincsearch noinfercase nois nojs nolbr nolisp noloadplugins nolz nomacatsui nomh nomod nomodifiable nomore nomousefocus nonu noodev nopaste nopreserveindent noprompt noreadonly noremap norevins norightleft nornu nors noruler nosc noscrollbind nosecure noshellslash noshiftround noshowcmd noshowmatch nosi nosmartcase nosmarttab nospell nosplitright nosr nosta nostmp noswf notagbsearch notagstack notbidi notermbidi notextauto notildeop notitle notop nottimeout noudf novb nowa nowb nowfh nowic nowildmenu nowinfixwidth nowmnu nowrapscan nowriteany nows
```
Comment (Pyrohh) [inline runtime/syntax/vim.vim]: This line has `nocp`, and there's `cp` somewhere else (just search `/\V\<cp\>`).

**Comment by Tranquility** [comment]
I implemented all comments and tagged it ready.

**Comment by ZyX-I** [comment]
Ref #2288.

**Comment by Pyrohh** [comment]
#2288 has been untouched for a while, so I'd like to merge this if no-one objects. Do you mind if I pick this up @Tranquility, or do you want to?

**Comment by Pyrohh** [comment]
#2276

**Pull request closed without merging**